catalyst/contrib/callbacks/draw_masks_callback.py

```diff
@@ -0,0 +1,205 @@
+from typing import Iterable, List, Optional, TYPE_CHECKING
+import os
+
+import numpy as np
+from skimage.color import label2rgb
+from skimage.color.colorlabel import DEFAULT_COLORS
+
+import torch
+
+from catalyst import utils
+from catalyst.callbacks import ILoggerCallback
+from catalyst.contrib.tools.tensorboard import SummaryWriter
+from catalyst.contrib.utils.cv.tensor import tensor_to_ndimage
+from catalyst.core.callback import CallbackNode, CallbackOrder
+
+if TYPE_CHECKING:
+    from catalyst.core.runner import IRunner
+
+DEFAULT_COLORS = np.array(DEFAULT_COLORS)
+
+
+class DrawMasksCallback(ILoggerCallback):
+    """
+    Logger callback draw masks for common segmentation task: image -> masks
+    """
+
+    def __init__(
+        self,
+        pred_mask_key: str,
+        image_key: Optional[str] = None,
+        gt_mask_key: Optional[str] = None,
+        mask2show: Optional[Iterable[int]] = None,
+        activation: Optional[str] = "Sigmoid",
+        log_name: str = "images",
+        summary_step: int = 50,
+        threshold: float = 0.5,
+    ):
+        """
+
+        Args:
+            pred_mask_key: predicted mask key
+            image_key: input image key. If None mask will be drawn on black
+            background
+            gt_mask_key: ground truth mask key. If None, will not be drawn
+            mask2show: mask indexes to show, if None all mask will be drawn. By
+            this parameter you can change the mask order
+            activation: An torch.nn activation applied to the outputs.
+            Must be one of ``'none'``, ``'Sigmoid'``, ``'Softmax2d'``
+            log_name: logging name. If you use several such "callbacks", they
+            must have different logging names
```

Comment:
why do you need it? by default, we usually have `tensorboard` logger already for each dataset
could you reuse it?
here is an example, https://github.com/catalyst-team/catalyst/blob/master/catalyst/contrib/callbacks/confusion_matrix_logger.py#L41

Reply:
I think that it is better to save images separately. Since they weigh a lot more than everything else, and sometimes there is a desire to remove them.

Reply:
The names are all debatable, if you want other names, I can change them.